pubsub/callback/callback.go

```diff
@@ -0,0 +1,41 @@
+package callback
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/micromdm/micromdm/pubsub"
+)
+
+type CallbackPublisher struct {
+	url  string
+	next pubsub.Publisher
+}
+
+func (p *CallbackPublisher) Publish(ctx context.Context, topic string, msg []byte) error {
+
+	// send to callback url
+	var buf bytes.Buffer
+	var event = struct {
+		Topic   string `json:"topic"`
+		Message []byte `json:"message"`
```

Comment:
this should marshal anything on the message bus to json, including the proto. The consumer might have to choose to decode the data on their own.